### PR TITLE
Add support for tabsets

### LINKED ISSUE #1667: Provide tab support for multiple languages
While reading the [telemetry](https://hexdocs.pm/telemetry/readme.html#usage) documentation. I noticed we code for two languages.

image

It would be cool if we can provide a tab selection for languages.

Don't have any idea how to do it. But we can get the blocks, order them by name, and show them like tabs.

image

With that, we can generate something like

```
Erlang | Elixir

telemetry:execute(
  [web, request, done],
  #{latency => Latency},
  #{request_path => Path, status_code => Status}
)
```

What do you guys think about that idea?

### PR DESCRIPTION
**Pull request by DavidOliver**
Closes #1667

Also:

* simplifies section heading markup and styling: section headings' innerText now directly usable as tab button labels
* tweaks code background color in dark theme to better match contrast of light theme

---

The readme has been updated to include a simple example tabset and show the necessary syntax.

Regarding the section heading changes, please note that I may not be completely familiar with all possible types of admonition block headings. I "fixed" all failing tests based on my expectations of the new markup, and the admonition blocks in the readme are displaying as expected, but you may wish to check this aspect in particular before merging.

Here's how the (unmodified by JavaScript) markup is displaying in my desktop EPUB reader:

Screenshot from 2023-05-04 14-24-13

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -211,6 +211,52 @@ For example, if you change the class name to `neutral`, you get the same admonit
...
+```
+
+---
+{: .tabs-close}
+
+Below is the example's source, showing the tabset syntax. `h3` headings indicate a new tab panel and are used as the tab button labels.
+
+````markdown
+---
+{: .tabs-open}
```
[the remark this answers is not in the document]
Comment (DavidOliver) [reply README.md:242]: If the attribute-setting is on the same line, I get `<p>--- {: .tabs-open}</p>` in the HTML.

File: README.md [hunk 2]
```diff
@@ -211,6 +211,52 @@ For example, if you change the class name to `neutral`, you get the same admonit
...
+### Erlang
+
+```erlang
+io:fwrite("Hello, world!\n").
+```
+
+---
+{: .tabs-close}
+
+Below is the example's source, showing the tabset syntax. `h3` headings indicate a new tab panel and are used as the tab button labels.
```
Comment (josevalim) [inline README.md:238]:
```suggestion
Below is the example's source, showing the tabset syntax. You must define all tabs between horizontal rules (`---`). The opening and closing rules must be followed by a newline and `{: .tags-open}` and `{: .tags-close}` respectively. `h3` headings indicate a new tab panel and are used as the tab button labels.
```

**Comment by josevalim** [comment]
Thank you @DavidOliver! It is a pity that `{: .tabs-close}` and `{: .tabs-open}` show up on GitHub but I can't think of any other options.

**Comment by josevalim** [comment]
Wait, could we use HTML comments for this? `<!-- tabs-open -->` and `<!-- tabs-close -->`? And then the JavaScript look for those nodes for splitting? Would this work?

**Comment by DavidOliver** [comment]
That seems like a good idea, assuming GitHub, etc., don't render HTML comments in rendered readmes.

I've just tried, and, currently, the comment is not making it to the HTML output. I wonder if we can change that...

**Comment by josevalim** [comment]
@DavidOliver i just pushed a commit to ExDoc so we preserve HTML comments from Markdown. Can you please give it a try? :)

**Comment by DavidOliver** [comment]
Sweet! I'll give it a go asap. Thanks!

**Comment by DavidOliver** [comment]
@josevalim Updated to use HTML comments. Thanks again.

**Comment by wojtekmach** [comment]
@DavidOliver this is amazing!

@josevalim would it make sense not to render comments in IEx, currently it looks like this:

image

**Comment by DavidOliver** [comment]
@wojtekmach, glad you like it.

I realised I have some updates to JS comments/documentation content to make. I'll push those a little later today.

File: README.md [hunk 3]
```diff
@@ -211,6 +211,50 @@ For example, if you change the class name to `neutral`, you get the same admonit
 >
 > This syntax will render an error block
 
+## Tabsets
+
+Where only one section of content of a series is likely to apply to the reader, you may wish to define a set of tabs.
+
+This example contains code blocks, separating them into tabs based on language:
+
+<!--  tabs-open -->
```
Comment (josevalim) [inline README.md:220]:
```suggestion
<!-- tabs-open -->
```

**Comment by josevalim** [comment]
Beautiful work! Sorry for the back and forth but I love where we landed! :green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**